scripts/src2dest.js

```diff
@@ -0,0 +1,209 @@
+import fs from 'fs'
+import readline from 'readline'
+
+// Not sure why calculated size is smaller than the actual one; this ratio is to adjust it to norm
+const adjustmentRatio = 1.2
+const targetChunkSize = 1e7
+const chunkSizeLimit = Math.round(targetChunkSize / adjustmentRatio)
+const encoder = new TextEncoder()
+const stringifyOptions = [null, 2] // Encoder, number of spaces
+
+const sourceInfo = {
+  path: '../src/',
+  fileName: 'cedict_ts',
+  fileExtension: 'u8'
+}
+
+const targetInfo = {
+  path: '../dest/',
+  fileName: 'cedict',
+  version: '',
+  fileExtension: 'json'
+}
+
+let counter = 1
+let metadataStr = ''
+let dictData = { // eslint-disable-line prefer-const
+  metadata: {},
+  entries: []
+}
+let sizeData = [] // eslint-disable-line prefer-const
+
+// An interface for reading from a source file
+const rlInterface = readline.createInterface({
+  input: fs.createReadStream(`${sourceInfo.path}${sourceInfo.fileName}.${sourceInfo.fileExtension}`),
+  output: process.stdout
+})
+
+rlInterface.on('line', (line) => {
+  if (/^#/.test(line)) {
+    // Assemble metadata into a single string for further parsing
+    metadataStr += `${line}\n`
+    return
+  }
+
+  const parts = line.match(/(\S+) (\S+) \[(.+)] \/(.+)\//)
```

Comment:
it looks like a new field in the data since we did Alpheios 1.0 is the addition of the classifiers following the definitions  e.g. (CL:座[zuo4],個|个[ge4]/). from the wiki it says "Classifiers follow the 'reference' syntax, are prefixed by 'CL:' and separated by a comma (no additional spacing)."

I think this would be a useful field to include in Alpheios. Chinese nouns that can be enumerated use different classifiers in between the numeral and the noun (e.g. in yi ge ren which means one person 'ge' is the classifier)

Can we add that when it's present in the data?

Reply:
I'm not sure if we want to do anything to try to further parse the entries with Middle Dot and commas. I could see a value in separating out names and proverbs into separate types of entries.  What do you think?

From the wiki

Middle dot
Middle dots are often used for separating western names:
珍・奧斯汀 珍・奥斯汀 [Zhen1 · Ao4 si1 ting1] /Jane Austen (1775-1817), English novelist/

A double width middle dot is used in the Chinese, a single width middle dot padded with spaces on both sides is used in the pinyin.

Comma
Commas are sometimes used in Chinese proverbs:
人為財死，鳥為食亡 人为财死，鸟为食亡 [ren2 wei4 cai2 si3 , niao3 wei4 shi2 wang2] /Human beings die in pursuit of wealth, and birds die in pursuit of food/…/

Reply:
Thanks for letting me know about these details. I do think that the deeper we parse the data, the better: if we'll need a bigger construct, it will be easier to assemble it from smaller parts. Will do the update tomorrow.

Reply:
In the example of a classifier given (`CL:座[zuo4],個|个[ge4]`) do I understand correctly that there are two classifiers separated by comma and we need to save them to the array? In that case for the first one `座` is a Chinese symbol (both traditional and simplified?) and `zuo4` is a pinyin? What would be the symbols for the second one? Is `個` the one for a traditional Chinese and `个` for a simplified one or is `個|个` a set of symbols for both?

Reply:
For names, would we split them in the first and the last and store both together with their pinyin?

Reply:
For proverbs, would we just mark entries as such but do no additional parsing of headwords and pinyin?

Reply:
yes, it looks to me like when the classifier has both a traditional and a simplified version, they are separated by | and when there is no | it is the same character for both traditional and simplified, and that multiple possible classifiers are separated by commas

so split on comma to get the list of classifiers, then split on | to get the traditional/simplified char for each (saving traditional to simplified if no simplified)

Reply:
> For names, would we split them in the first and the last and store both together with their pinyin 
yes, I think that makes sense

Reply:
>For proverbs, would we just mark entries as such but do no additional parsing of headwords and pinyin?

Not sure what to do with proverbs exactly. maybe treat them the same as the other entries, but with a separate field to indicate that it's a proverb?

Reply:
Some definitions has classifiers in the middle. Would it be correct just to cut them out (and store in their own fields, of course), so that `'drum/CL:通[tong4],面[mian4]/to drum/to strike/to rouse/to bulge/to swell'` would become `'drum/to drum/to strike/to rouse/to bulge/to swell'`?

Reply:
The related question: do I understand correctly that a slash in definitions like above (`drum/to drum/to strike/to rouse/to bulge/to swell`) separate different definitions (or meanings maybe if they're not the same thing). Then, if we're already goes heavy on parsing, maybe it would make sense to parse definitions like that into an array of strings?

Reply:
yes, that does make sense

Reply:
>Some definitions has classifiers in the middle. Would it be correct just to cut them out (and store in their own fields, of course), so that 'drum/CL:通[tong4],面[mian4]/to drum/to strike/to rouse/to bulge/to swell' would become 'drum/to drum/to strike/to rouse/to bulge/to swell'?

That's interesting because in this case at least, it indicates multiple parts of speech.  I don't know how regular that is in the data, but I would be tempted to split that entry into two, since it's really a homonym then.